From 4053ef436d6cf1e538f23ad95c55558b2a5f9a7f Mon Sep 17 00:00:00 2001
From: Salini Venate <salini.venate@intel.com>
Date: Mon, 26 Feb 2024 11:16:03 +0530
Subject: [PATCH] WA - Fix StrictMode memory leak error in Webview

When the device goes to S3 state and wakes up
below error is seen and the browser exits:
StrictMode: StrictMode policy violation:
android.os.strictmode.InstanceCountViolation:
class org.chromium.webview_shell.WebViewBrowserActivity;
instances=4; limit=3

Disabled the StrictMode memory leak check for webview browser.

Tracked-On: OAM-114158
Signed-off-by: Salini Venate <salini.venate@intel.com>
---
 .../chromium/webview_shell/WebViewBrowserActivity.java    | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/org/chromium/webview_shell/WebViewBrowserActivity.java b/src/org/chromium/webview_shell/WebViewBrowserActivity.java
index 0a7b637..a507868 100644
--- a/src/org/chromium/webview_shell/WebViewBrowserActivity.java
+++ b/src/org/chromium/webview_shell/WebViewBrowserActivity.java
@@ -249,10 +249,10 @@ public class WebViewBrowserActivity extends Activity implements PopupMenu.OnMenu
         // Conspicuously omitted: detectCleartextNetwork() and detectFileUriExposure() to permit
         // http:// and file:// origins.
         StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
-                .detectActivityLeaks()
-                .detectLeakedClosableObjects()
-                .detectLeakedRegistrationObjects()
-                .detectLeakedSqlLiteObjects()
+                //.detectActivityLeaks()
+                //.detectLeakedClosableObjects()
+                //.detectLeakedRegistrationObjects()
+                //.detectLeakedSqlLiteObjects()
                 .penaltyLog()
                 .penaltyDeath()
                 .build());
-- 
2.17.1

